sydent/http/httpclient.py

```diff
@@ -18,51 +18,60 @@
 import logging
 
 from StringIO import StringIO
-from twisted.internet import defer, reactor, ssl
-from twisted.internet.endpoints import HostnameEndpoint, wrapClientTLS
-from twisted.internet._sslverify import _defaultCurveName, ClientTLSOptions
+from twisted.internet import defer, reactor
 from twisted.web.client import FileBodyProducer, Agent, readBody
 from twisted.web.http_headers import Headers
-import twisted.names.client
-from twisted.names.error import DNSNameError
-from OpenSSL import SSL, crypto
+from sydent.http.matrixfederationagent import MatrixFederationAgent
+
+from sydent.utils.tls import ClientTLSOptionsFactory
 
 logger = logging.getLogger(__name__)
 
-class SimpleHttpClient(object):
-    """
-    A simple, no-frills HTTP client based on the class of the same name
-    from synapse
+class HTTPClient(object):
+    """A base HTTP class that contains methods for making GET and POST HTTP
+    requests.
     """
-    def __init__(self, sydent, endpoint_factory=None):
-        self.sydent = sydent
-        if endpoint_factory is None:
-            # The default endpoint factory in Twisted 14.0.0 (which we require) uses the
-            # BrowserLikePolicyForHTTPS context factory which will do regular cert validation
-            # 'like a browser'
-            self.agent = Agent(
-                reactor,
-                connectTimeout=15,
-            )
-        else:
-            self.agent = Agent.usingEndpointFactory(
-                reactor,
-                endpoint_factory,
-            )
-
     @defer.inlineCallbacks
     def get_json(self, uri):
+        """Make a GET request to an endpoint returning JSON and parse result
+
+        :param uri: The URI to make a GET request to.
+        :type uri: str
+        :returns a deferred containing JSON parsed into a Python object.
+        :rtype: Deferred[any]
+        """
         logger.debug("HTTP GET %s", uri)
 
         response = yield self.agent.request(
             "GET",
             uri.encode("ascii"),
         )
         body = yield readBody(response)
-        defer.returnValue(json.loads(body))
+        try:
+            json_body = json.loads(body)
+        except Exception as e:
+            logger.exception("Error parsing JSON from %s", uri)
+            raise e
```

Comment:
A bare `raise` is generally preferable as it means the stack trace will include the lower stack frames (ie, `json.loads` and its descendents in this case). `raise e` will start the stack trace from the raise point and any information about lower stack frames is lost.

```suggestion
            raise
```

Reply:
Good to know, thanks.